clang/lib/Sema/SemaTemplateInstantiate.cpp

```diff
@@ -3430,6 +3430,7 @@ Sema::InstantiateClass(SourceLocation PointOfInstantiation,
     llvm::raw_string_ostream OS(Name);
```

Comment:
Should we do this for other things?
Most of the nodes are likely to have a source location they can be attributed to.

Reply:
I think we can do this on a need basis. Most source actions (like Parse*) already have this information (although not structured). See the unit test for examples.